contracts/modules/cover/CoverViewer.sol

```diff
@@ -111,6 +111,17 @@ contract CoverViewer {
     return coverViews;
   }
 
+  function getCoverSegments(uint coverId) external view returns (CoverSegment[] memory segments) {
```

Comment:
@mixplore is this enough for your usecase? or should we implement this to take as input an array of cover IDs, and return a mapping coverID -> segments?

Reply:
🤔  Now that you're mentioning, yeah, it would probably be better to be able to get the segments for multiple covers. I think we'll need at least the active segment of the cover (with period and amount details) in order to display the data in `My covers` table. The way I see it coming from an API is described in this ticket [#4](https://github.com/NexusMutual/cover-router/issues/4) if that helps.

Reply:
understood, DMing about this

Reply:
after offline discussions, we're keeping this implementation as is.